internal/pkg/manifest/validate.go

```diff
@@ -1774,6 +1775,9 @@ func validateExposedPorts(opts validateExposedPortsOpts) error {
 	if err := populateALBPortsAndValidate(containerNameFor, opts); err != nil {
 		return err
 	}
+	if err := populateNLBPortsAndValidate(containerNameFor, opts); err != nil {
```

Comment:
quick question: what is the responsibility of `validateExposedPorts`?
1. validate that no port is exposed by two containers
2. validate that the protocol is valid
Is this the complete list of responsibilities?

Reply:
Yes making sure that we error out when customer defines duplicate ports. 
Validating protocol is extra thing that I was doing here which is already being done in the validate NLB. I will remove that code in the next commit. 

Is there anything else you would want me to take care as a part of `validateExposedPorts`?

Reply:
no no I think it makes sense that "1. validate duplicate ports" is the sole purpose of `validateExposedPorts`. In fact, in addition to NLB, what do you think of moving the protocol check in `validateSidecarExposedPorts` to sidecar level as well? If I understand the design correctly,  you don't need any service-level information in order to know a protocol is invalid. 

If you do decide to do this, don't worry about adding it to this PR! We can do it in a follow-up PR.

Reply:
Done